tidy3d/components/simulation.py

```diff
@@ -1196,11 +1196,19 @@ def monitors_data_size(self) -> Dict[str, float]:
         return data_size
 
     def _monitor_num_cells(self, monitor: Monitor) -> int:
-        """Total number of cells included by monitor based on simulation grid."""
-        num_cells = self.discretize_monitor(monitor).num_cells
-        # take monitor downsampling into account
-        num_cells = monitor.downsampled_num_cells(num_cells)
-        return np.prod(np.array(num_cells, dtype=np.int64))
+        """Total number of cells included in monitor based on simulation grid."""
+        if isinstance(monitor, SurfaceIntegrationMonitor):
+            monitors = monitor.integration_surfaces
+        else:
+            monitors = [monitor]
+
+        num_cells_total = 0
+        for mnt in monitors:
+            num_cells = self.discretize_monitor(mnt).num_cells
+            # take monitor downsampling into account
```

Comment:
Maybe I'd consider separating the logic for computing the number of cells for a regular monitor from the logic for computing the number of cells for a `SurfaceIntegrationMonitor`.

```py
def _num_cells_vol_monitor(self, monitor: Monitor) -> int:
    """Get the number of measurement cells in a monitor given the simulation grid and downsampling."""
    sub_grid = self.discretize_monitor(monitor)
    num_cells = sub_grid.num_cells
    num_cells_downsampled = monitor.downsampled_num_cells(num_cells)
    return np.prod(np.array(num_cells, dtype=np.int64))

def _num_cells_monitor(self, monitor: Monitor) -> int:
    if isinstance(monitor, SurfaceIntegrationMonitor):
        return sum([self._num_cells_vol_monitor(mnt) for mnt in monitor.integration_surfaces])
    return self._num_cells_vol_monitor(monitor)
```

I think it's a bit better to separate the logic out like this, what do you think?

Reply:
How's this, I left the function as internal to `_num_cells_monitor`?

By the way merging is now blocked if changes are requested, so if things look good you need to somehow mark that they've been resolved. (I can override, but better to do it properly)

Reply:
that sounds fine (if you think it wouldn't be useful elsewhere) thanks, resolved

Reply:
I still get this

![image](https://github.com/flexcompute/tidy3d/assets/92756559/99e31566-1fe8-42b2-add3-b10bad9fc700)